commit 7ed8c4ad364bccfb5aa72d63562be45a95f5ea2f (refs/changes/71/3371/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-02-15T11:53:27-08:00 (1 year, 8 months ago)
    
    TOOLS-1986 MG's ./tools/ls-missing-release-builds will falsely report missing builds if mls env isn't properly setup
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/tools/ls-missing-release-builds b/tools/ls-missing-release-builds
index 809dc9a..3b1d3f0 100755
--- a/tools/ls-missing-release-builds
+++ b/tools/ls-missing-release-builds
@@ -71,6 +71,10 @@ if [[ -z "$targets" ]]; then
         | grep -v '^headnode-joyent-debug ')
 fi
 
+# Ensure mls is setup properly at all.
+mls /Joyent_Dev/public/builds >/dev/null \
+    || fatal "cannot list /Joyent_Dev/public/builds"
+
 echo "$targets" | while read target public; do
     if [[ "$public" == "true" ]]; then
         latest_mpath=/Joyent_Dev/public/builds/$target/$release-latest
